// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";


package userlogin;

// The userlogin service definition.
service User {
  // sign in
  rpc AddUser (UserInfo) returns (AddReply) {}
  rpc LoginIn (UserInfo) returns (LoginReply) {}
  rpc IsLogin (UserLoginInfo) returns (IsReply) {}
}

// The request message containing the user's name and how many greetings
// they want.
message UserInfo {
  string name = 1;
  string password = 2;
}

// A response message containing a greeting
message AddReply {
  string rscode = 1;
  string message = 2;
}


message LoginReply {
  string rscode = 1;
  string message = 2;
  string handlecode = 3;
}

message UserLoginInfo {
  string name = 1;
  string handlecode = 2;
}

message IsReply {
  string message = 1;
}
